vm/vm_test.go

```diff
@@ -136,7 +136,8 @@ func (r *TestResolver) GetFunction(module, name string) HostFunction {
 
 func TestVmError(t *testing.T) {
 	tests := []vmTest{
-		{name: "exit", entry: "calc", params: []uint64{}, expectedError: "unreachable"},
+		{name: "exit", entry: "calc", params: []uint64{1}, expectedErr: ErrUnreachable},
+		{name: "local", entry: "calc", params: []uint64{}, expectedErr: ErrInvalidParamNumber},
 	}
 	for _, test := range tests {
 		defer func() {
```

Comment:
defer in this usecase is not going to work as intended inside a loop
https://github.com/golang/go/issues/3978

Reply:
This can be easily fixed by wrapping each test case inside `t.Run()`